package/MDAnalysis/analysis/base.py

```diff
@@ -407,49 +535,88 @@ def run(self, start=None, stop=None, step=None, frames=None,
-        
+
+        .. versionadded:: 2.5.0
         progressbar_kwargs : dict, optional
-            ProgressBar keywords with custom parameters regarding progress bar position, etc; 
+            ProgressBar keywords with custom parameters regarding progress bar position, etc;
             see :class:`MDAnalysis.lib.log.ProgressBar` for full list.
-
+            Available only for `backend='serial'`
+        backend : Union[str, BackendBase], optional
+            By default, performs calculations in a serial fashion.
+            Otherwise, user can choose a backend:
+               - `str` is matched to a builtin backend (one of `serial`, `multiprocessing` and `dask`)
+               - `BackendBase` subclass is checked for the presence of `apply` method
+        n_workers : int
+            positive integer with number of workers (processes, in case of built-in backends) to split the work between
+        n_parts : int, optional
+            number of parts to split computations across. Can be more than number of workers,
+            especially if you want to get a progressbar with `dask.distributed` backend.
+        unsafe : bool, optional
+            if you want to run your custom backend on a parallelizable class that has not been tested by developers, by default False
 
         .. versionchanged:: 2.2.0
             Added ability to analyze arbitrary frames by passing a list of
             frame indices in the `frames` keyword argument.
 
         .. versionchanged:: 2.5.0
-            Add `progressbar_kwargs` parameter, 
+            Add `progressbar_kwargs` parameter,
             allowing to modify description, position etc of tqdm progressbars
         """
-        logger.info("Choosing frames to analyze")
-        # if verbose unchanged, use class default
-        verbose = getattr(self, '_verbose',
-                          False) if verbose is None else verbose
+        # default to serial execution
+        backend = "serial" if backend is None else backend
 
-        self._setup_frames(self._trajectory, start=start, stop=stop,
-                           step=step, frames=frames)
-        logger.info("Starting preparation")
+        if progressbar_kwargs and backend != "serial":
+            raise ValueError("Can not display progressbar with non-serial backend")
+
+        n_workers = 1 if n_workers is None else n_workers
+
+        # set n_parts and check that is has a reasonable value
+        n_parts = n_workers if n_parts is None else n_parts
+
+        # do this as early as possible to check client parameters before any computations occur
+        executor = self._configure_backend(backend=backend, n_workers=n_workers, unsafe=unsafe)
+        if (
+            hasattr(executor, "n_workers") and n_parts < executor.n_workers
+        ):  # using executor's value here for non-default executors
+            warnings.warn(f"likely running not at full capacity: {executor.n_workers=} is greater than {n_parts=}")
```

Comment:
Please check the Codecov warnings and add the appropriate test.

```suggestion
            warnings.warn(f"Analysis not making use of all workers: {executor.n_workers=} is greater than {n_parts=}")
```

Reply:
Fixed in commit.